src/BasicUtility.cpp

```diff
@@ -0,0 +1,38 @@
+#include "BasicUtility.h"
+
+#include "libmesh/dof_map.h"
+#include "libmesh/elem.h"
+#include "libmesh/libmesh.h"
+#include "libmesh/linear_implicit_system.h"
+#include "libmesh/numeric_vector.h"
+#include "libmesh/system.h"
+
+void PopulateSyntheticData(libMesh::EquationSystems &equation_system,
+                           const std::string system_name,
+                           const std::string variable_name,
+                           double (*CalculateData)(const libMesh::Point &),
+                           bool needs_re_init) {
+
+  libMesh::MeshBase &mesh = equation_system.get_mesh();
+  libMesh::LinearImplicitSystem &system =
+      equation_system.add_system<libMesh::LinearImplicitSystem>(system_name);
+
+  system.add_variable(variable_name, libMesh::CONSTANT, libMesh::MONOMIAL);
+  if (needs_re_init) {
```

Comment:
Instead of passing this Boolean, why not
```suggestion
  if (system.is_initialized()) {
```